packages/expo-auth-session/src/SessionUrlProvider.ts

```diff
@@ -1,22 +1,118 @@
-import { Platform } from '@unimodules/core';
+import { Platform } from '@unimodules/react-native-adapter';
 import Constants, { ExecutionEnvironment } from 'expo-constants';
+import * as Linking from 'expo-linking';
+import { CreateURLOptions } from 'expo-linking';
+import { resolveScheme } from 'expo-linking/build/Schemes';
+import qs, { ParsedQs } from 'qs';
 
-import { BareSessionUrlProvider } from './BareSessionUrlProvider';
-import { ManagedSessionUrlProvider } from './ManagedSessionUrlProvider';
+const { manifest } = Constants;
 
-export interface SessionUrlProvider {
-  getDefaultReturnUrl: () => string;
-  getStartUrl: (authUrl: string, returnUrl: string) => string;
-  getRedirectUrl: (urlPath?: string) => string;
-}
+export class SessionUrlProvider {
+  private static readonly BASE_URL = `https://auth.expo.io`;
+  private static readonly SESSION_PATH = 'expo-auth-session';
+
+  getDefaultReturnUrl(urlPath?: string, options?: Omit<CreateURLOptions, 'queryParams'>): string {
+    const queryParams = SessionUrlProvider.getHostAddressQueryParams();
+    let path = SessionUrlProvider.SESSION_PATH;
+    if (urlPath) {
+      path = [path, SessionUrlProvider.removeLeadingSlash(urlPath)].filter(Boolean).join('/');
+    }
+
+    return Linking.createURL(path, {
+      // The redirect URL doesn't matter for the proxy as long as it's valid, so silence warnings if needed.
+      scheme: options?.scheme ?? resolveScheme({ isSilent: true }),
+      queryParams,
+      isTripleSlashed: options?.isTripleSlashed,
+    });
+  }
+
+  getStartUrl(authUrl: string, returnUrl: string): string {
+    if (Platform.OS === 'web' && !Platform.isDOMAvailable) {
+      // Return nothing in SSR envs
+      return '';
+    }
+    const queryString = qs.stringify({
+      authUrl,
+      returnUrl,
+    });
 
-export function getSessionUrlProvider(): SessionUrlProvider {
-  if (
-    (Constants.executionEnvironment === ExecutionEnvironment.Standalone ||
-      Constants.executionEnvironment === ExecutionEnvironment.StoreClient) &&
-    Platform.OS !== 'web'
-  ) {
-    return new ManagedSessionUrlProvider();
+    return `${this.getRedirectUrl()}/start?${queryString}`;
+  }
+
+  getRedirectUrl(urlPath?: string): string {
+    if (Platform.OS === 'web') {
+      if (Platform.isDOMAvailable) {
+        return [window.location.origin, urlPath].filter(Boolean).join('/');
+      } else {
+        // Return nothing in SSR envs
+        return '';
+      }
+    }
+
+    const legacyExpoProjectId = manifest.currentFullName || manifest.id;
+
+    if (!legacyExpoProjectId) {
+      let nextSteps = '';
+      if (__DEV__) {
+        if (Constants.executionEnvironment === ExecutionEnvironment.Bare) {
+          nextSteps =
+            ' Please ensure you have the latest version of expo-constants installed and rebuild your native app. You can verify that currentFullName is defined by running `expo config --type public | grep currentFullName`';
```

Comment:
```suggestion
            ' Please ensure you have the latest version of expo-constants installed and rebuild your native app. You can verify that currentFullName is defined by running `expo config --type public` and inspecting the output.';
```

because windows